src/core/Akka.Tests/Loggers/LogFilterEvaluatorSpecs.cs

```diff
@@ -0,0 +1,224 @@
+using Akka.Actor;
+using Akka.Actor.Setup;
+using Akka.Configuration;
+using Akka.Event;
+using Akka.TestKit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Tests.Loggers;
+
+/// <summary>
+/// Goal of these specs are mostly to make sure our default <see cref="Regex"/>s are working as expected
+/// </summary>
+// ReSharper disable once ClassNeverInstantiated.Global
+public class LogFilterEvaluatorSpecs
+{
+    public class LogFilterSetupSpecs : AkkaSpec
+    {
+        public static Setup LoggerSetup()
+        {
+            var builder = new LogFilterBuilder();
+            builder.ExcludeSourceContaining("Akka.Tests")
+                .ExcludeMessageContaining("foo-bar");
+            return builder.Build();
+        }
+        
+        public static ActorSystemSetup CustomLoggerSetup()
+        {
+            var hocon = @$"akka.stdout-logger-class = ""{typeof(CustomLogger).AssemblyQualifiedName}""";
+            var bootstrapSetup = BootstrapSetup.Create().WithConfig(ConfigurationFactory.ParseString(hocon));
+            return ActorSystemSetup.Create(bootstrapSetup, LoggerSetup());
+        }
+        
+        // create a custom MinimalLogger that subclasses StandardOutLogger
+        public class CustomLogger : StandardOutLogger
+        {
+            protected override void Log(object message)
+            {
+                if (message is LogEvent e)
+                {
+                    if (Filter.ShouldTryKeepMessage(e, out _))
+                    {
+                        _events.Add(e);
+                    }
+                }
+               
+            }
+            
+            private readonly List<LogEvent> _events = new();
+            public IReadOnlyList<LogEvent> Events => _events;
+        }
+    
+        public LogFilterSetupSpecs(ITestOutputHelper output) : base(CustomLoggerSetup(),
+            output: output)
+        {
+            _logger = (CustomLogger)Sys.Settings.StdoutLogger;
+        }
+
+        private readonly CustomLogger _logger;
+        
+        [Fact]
+        public void LogFilterEnd2EndSpec()
+        {
+            // subscribe to warning level log events
+            Sys.EventStream.Subscribe(TestActor, typeof(Warning));
+            
+            // produce three warning messages - that hits the source filter, another that hits the message filter, and a third that hits neither
+            var loggingAdapter1 = Logging.GetLogger(Sys, "Akka.Tests.Test1");
+            var loggingAdapter2 = Logging.GetLogger(Sys, "Akka.Util.Test2");
+            
+            // should be filtered out based on Source
+            loggingAdapter1.Warning("foo-bar");
+            
+            // should be filtered out based on message content
+            loggingAdapter2.Warning("foo-bar");
+            
+            // should be allowed through
+            loggingAdapter2.Warning("baz");
+            
+            // expect only the last message to be received
+            ReceiveN(3);
```

Comment:
Log events will still hit the `EventStream` - the filter runs at the layer beneath this. **AND THAT IS IMPORTANT**: if we have the log filter layer running at this layer, at the point where `LogEvent`s are EMITTED, this WILL KILL PERFORMANCE. 

Instead, we make the decision about whether or not to keep / include a log at the layer right before the log is rendered. **This is the correct place to do this**.

Reply:
Anyway, that's why we still receive all 3 log events.